### PR TITLE
Display field description

### LINKED ISSUE #157: Display field description in the UI
Show the field's description from the schema in the UI.

### PR DESCRIPTION
**Pull request by psrpinto**
Fixes #157 

This PR displays the description of fields, coming from `schema.json`. I also took the opportunity to simplify some places of the code related to handling fields.

## Screen capture

Screenshot 2024-12-17 at 17 08 15

### CODE REVIEW

File: src/ui/components/FieldsEditor/FieldsEditor.tsx
Comment (psrpinto) [inline src/ui/components/FieldsEditor/FieldsEditor.tsx]: I haven't done this here to not conflict with #170. Once we merge that one I will take care of this.

**Comment by psrpinto** [comment]
I will wait for #170 to be merged, and adapt this one as needed.

File: src/ui/components/FieldsEditor/FieldsEditor.tsx
Comment (ashfame) [reply src/ui/components/FieldsEditor/FieldsEditor.tsx]: Everything looks good in the PR! This comment tripped me up though: `Filter out fields that are not in the schema.`

Why would a subject have fields that are not in the schema? Shouldn't schema already be driving that aspect as well? Or I misunderstood?

Comment (psrpinto) [reply src/ui/components/FieldsEditor/FieldsEditor.tsx]: Great question, it's not obvious when this would happen. This would only be the case when a field is removed from the schema, basically the scenario described in #159. But actually I think this has already been addressed in #170, AFAICT fields that are no longer in the schema will also not be present in the subject that is returned from `fromApiResponse()`. So I will remove this code and the TODO comment.

**Comment by psrpinto** [comment]
I force-pushed to fix conflicts. Only the last commit is new.

**Review by ashfame — APPROVED** [review]
🚀

**Pull request merged**
